regex-capi/include/rure.h

```diff
@@ -567,6 +567,22 @@ void rure_error_free(rure_error *err);
  */
 const char *rure_error_message(rure_error *err);
 
+/*
+ * rure_escape returns a NULL terminated string where all meta characters have
+ * been escaped.
+ *
+ * The pointer returned must not be freed directly. Instead, it should be freed
+ * by calling rure_string_free.
+ */
+const char *rure_escape(const char *pattern, size_t length);
```

Comment:
Hmm I don't think this signature makes sense to me. It's taking a `char *` and _also_ a length. Usually a `char *` is a NUL terminated string, which means the length isn't necessary. If you want to use a length with the string, then you probably want `uint8_t *` instead. However, I think regex patterns tend to be NUL terminated string literals, so I'd just drop the length parameter here.

Also, patterns are required to be UTF-8, so this needs to be able to return an error.